.travis.yml

```diff
@@ -12,78 +12,89 @@ os:
       #- SETUP=minimal PYTHON_VERSION=3.3
       #- SETUP=full PYTHON_VERSION=3.3
+
 matrix:
   fast_finish: true
   # allow_failures:
       #- env: SETUP=minimal PYTHON_VERSION=3.3
       #- env: SETUP=full PYTHON_VERSION=3.3
+  include:
+    - os: linux
+      env: MAIN_CMD="make -C ./package/doc/sphinx clean html"
+           BUILD_DOCS=true
+           SETUP=full
+
+    - os: linux
+      env: MAIN_CMD="pylint --rcfile=package/.pylintrc package/MDAnalysis"
 
-before_install:
-  - if [[ $TRAVIS_OS_NAME == 'linux' ]]; then wget http://repo.continuum.io/miniconda/Miniconda-latest-Linux-x86_64.sh -O miniconda.sh; fi
-  - chmod +x miniconda.sh
-  - ./miniconda.sh -b -p $(pwd)/miniconda
+    - os: linux
+      env: MAIN_CMD="pylint --rcfile=package/.pylintrc testsuite/MDAnalysisTests"
+
+install:
+  - wget http://repo.continuum.io/miniconda/Miniconda-latest-Linux-x86_64.sh -O miniconda.sh
+  - bash miniconda.sh -b -p $(pwd)/miniconda
   - export PATH=$(pwd)/miniconda/bin:$PATH
   - conda config --add channels MDAnalysis
   - conda config --add channels conda-forge
-  - conda update --yes conda
-  - conda install --yes pylint
-install:
-  # Minimal installation!
-  -  conda create --yes -q -n pyenv python=$PYTHON_VERSION numpy mmtf-python nose=1.3.7 mock sphinx=1.3 six biopython networkx cython joblib
+  - conda config --set always_yes yes
+  - conda create -q -n pyenv python=$PYTHON_VERSION
   - source activate pyenv
+  # Minimal installation!
+  - conda install numpy mmtf-python nose=1.3.7 mock six biopython networkx cython joblib pylint
   # Install griddataformats from PIP so that scipy is only installed in the full build (#1147)
-  - pip install griddataformats
-  - pip install nose-timer
+  - pip install griddataformats nose-timer
   # Add missing packages for the full build
   - |
     if [[ $SETUP == 'full' ]]; then \
-        conda install --yes matplotlib netcdf4 scikit-learn scipy seaborn coveralls; \
-        conda install -c biobuilds --yes clustalw=2.1; \
+        conda install matplotlib netcdf4 scikit-learn scipy seaborn coveralls; \
+        conda install -c biobuilds clustalw=2.1; \
+        # additional external tools (Issue #898) -- HOLE
+        bash ./maintainer/install_hole.sh $TRAVIS_OS_NAME "${HOME}/${MDA_OPTPACKAGES}"; \
+        HOLE_BINDIR="${HOME}/${MDA_OPTPACKAGES}/hole2/exe"; \
+        export PATH=${PATH}:${HOLE_BINDIR}; \
     fi
-  # additional external tools (Issue #898) -- HOLE
-  - |
-     if [[ $SETUP == 'full' ]]; then \
-         bash ./maintainer/install_hole.sh $TRAVIS_OS_NAME "${HOME}/${MDA_OPTPACKAGES}"; \
-         HOLE_BINDIR="${HOME}/${MDA_OPTPACKAGES}/hole2/exe"; \
-         export PATH=${PATH}:${HOLE_BINDIR}; \
-     fi
   # ensure that cython files are rebuilt
   - find . -name '*.pyx' -exec touch '{}' \;
-  # Install MDAnalysis
-  - pip install -v package/
-  - pip install testsuite/
+  # Install MDAnalysis based on build setup
+  - |
+    if [[ $BUILD_DOCS == "TRUE" ]] ; then \
```

Comment:
this part should select what to install based on the build environment. But it isn't working. I'm always installing via pip